clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -270,6 +275,27 @@ protected RecordSerializer<K> getKeySerializerForRequest() {
         return getKeySerializerWithRetryWithShortInterval();
       } catch (Exception e) {
         whetherStoreInitTriggeredByRequestFail = true;
+        // Kick off an async thread to keep retrying
+        asyncStoreInitThread = new Thread(() -> {
+          while (true) {
```

Comment:
Let's check `Thread.currentThread().isInterrupted()` here in case the exception is handled but not rethrown.

Reply:
Is it possible? The code will rethrow the exception if it gets interrupted.

Reply:
Yes, someone in `getKeySerializerWithRetryWithShortInterval` down the call-chain may handle interrupted exception. Then we'll be spinning in this loop forever. It's a good practice to check this flag in top-level spin-loops.

Reply:
The thread logic is purely inside this function, so it is not possible other function catches the exception and not exit.
But I got your point and I can add the check in the while loop.

Reply:
Is it possible since all the logic is embedded in this function? since the rethrown exception will break the logic.

Reply:
Also if the async store init succeeds, the loop will break as well.